FitTheWall/Assets/ThreeDPoseUnityBarracuda/Scripts/VideoCapture.cs

```diff
@@ -0,0 +1,125 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Video;
+
+public class VideoCapture : MonoBehaviour
+{
+    public GameObject InputTexture;
+    public RawImage VideoScreen;
+    public GameObject VideoBackground;
+    public float VideoBackgroundScale;
+    public LayerMask _layer;
+    public bool UseWebCam = true;
+    public int WebCamIndex = 0;
+    public VideoPlayer VideoPlayer;
+
+    private WebCamTexture webCamTexture;
+    private RenderTexture videoTexture;
+
+    private int videoScreenWidth = 2560;
+    private int bgWidth, bgHeight;
+
+    public RenderTexture MainTexture { get; private set; }
+
+    /// <summary>
+    /// Initialize Camera
+    /// </summary>
+    /// <param name="bgWidth"></param>
+    /// <param name="bgHeight"></param>
+    public void Init(int bgWidth, int bgHeight)
+    {
+        this.bgWidth = bgWidth;
+        this.bgHeight = bgHeight;
+        if (UseWebCam) CameraPlayStart();
+        else VideoPlayStart();
+    }
+
+    /// <summary>
+    /// Play Web Camera
+    /// </summary>
+    private void CameraPlayStart()
+    {
+        WebCamDevice[] devices = WebCamTexture.devices;
+        if(devices.Length <= WebCamIndex)
+        {
+            WebCamIndex = 0;
+        }
+        
+        webCamTexture = new WebCamTexture(devices[WebCamIndex].name);
+
+        var sd = VideoScreen.GetComponent<RectTransform>();
+        VideoScreen.texture = webCamTexture;
+
+        webCamTexture.Play();
+
+        sd.sizeDelta = new Vector2(videoScreenWidth, videoScreenWidth * webCamTexture.height / webCamTexture.width);
+        var aspect = (float)webCamTexture.width / webCamTexture.height;
+        VideoBackground.transform.localScale = new Vector3(aspect, 1, 1) * VideoBackgroundScale;
+        VideoBackground.GetComponent<Renderer>().material.mainTexture = webCamTexture;
+
+        InitMainTexture();
+    }
+
+    /// <summary>
+    /// Play video
+    /// </summary>
+    private void VideoPlayStart()
+    {
+        videoTexture = new RenderTexture((int)VideoPlayer.clip.width, (int)VideoPlayer.clip.height, 24);
+
```

Comment:
functions that contain "more logic", should contain comments to explain what kind of actions are taken